stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -260,11 +303,21 @@ public struct Unsafe${Mutable}BufferPointer<Element>
         base: self, bounds: bounds)
     }
 %  if Mutable:
-    set {
+    nonmutating set {
       _debugPrecondition(bounds.lowerBound >= startIndex)
       _debugPrecondition(bounds.upperBound <= endIndex)
+      _debugPrecondition(bounds.count == newValue.count)
```

Comment:
I think this one should be a full `_precondition` (similar to the one found in `_writeBackMutableSlice`).

Reply:
The analog line in UnsafeRawBufferPointer uses a `_debugPrecondition`: https://github.com/apple/swift/blob/ffd34239a1b6f046346769a55ae8e717f64e13eb/stdlib/public/core/UnsafeRawBufferPointer.swift.gyb#L468
Is there a reason to make it stronger in the typed buffers, or should it be a `_precondition` in both places?

Reply:
The reasoning behind `_debugPrecondition` is that this is already an `Unsafe` type and being at the the lowest level of abstraction will be used in performance critical code.